pkg-manager/core/src/install/index.ts

```diff
@@ -566,14 +572,31 @@ export async function mutateModules (
         optionalDependencies,
         updateWorkspaceDependencies: project.update,
         preferredSpecs,
-        saveCatalogName: opts.saveCatalogName,
+        saveCatalogName,
         overrides: opts.overrides,
         defaultCatalog: opts.catalogs?.default,
       })
+
       projectsToInstall.push({
         pruneDirectDependencies: false,
         ...project,
-        wantedDependencies: wantedDeps.map(wantedDep => ({ ...wantedDep, isNew: !currentBareSpecifiers[wantedDep.alias], updateSpec: true, nodeExecPath: opts.nodeExecPath })),
+        wantedDependencies: wantedDeps.map(wantedDep => {
+          if (saveCatalogName && !wantedDep.bareSpecifier.includes(':')) {
```

Comment:
What's `!wantedDep.bareSpecifier.includes(':')` checking for? If you're checking whether `wantedDep.bareSpecifier` is not a semver range, could we use the `semver` library for that?